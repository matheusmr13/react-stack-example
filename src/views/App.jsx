import React from 'react';
import Playlists from 'views/playlists';

import './App.css';

const App = () => (
	<div className="App">
		<header className="App-header">
			<h1 className="App-title">Welcome to Spotifood</h1>
		</header>
		<Playlists />
	</div>
);

export default App;
