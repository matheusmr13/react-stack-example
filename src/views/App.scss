.App {
	text-align: center;

	&-header {
		background-color: #222;
		padding: 20px;
		color: white;
	}
	&-title {
		font-size: 1.5em;
	}
	&-intro {
		font-size: large;
	}
}