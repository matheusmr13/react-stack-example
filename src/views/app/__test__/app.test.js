import React from 'react';
import { shallow } from 'enzyme';
import App from 'views/app';

describe('App component', () => {
	const fetchLoggedUser = jest.fn();
	const app = shallow(<App
		loggedUser={null}
		fetchLoggedUser={fetchLoggedUser}
	/>);
	it('should render properly', () => {
		expect(fetchLoggedUser).toHaveBeenCalled();
		expect(app).toMatchSnapshot();
	});
	it('should open drawer', () => {
		app.instance().onOpenAppBar();
		expect(app.state().isFilterOpen).toBeTruthy();
	});
	it('should close drawer', () => {
		app.instance().onCloseAppBar();
		expect(app.state().isFilterOpen).toBeFalsy();
	});
});
