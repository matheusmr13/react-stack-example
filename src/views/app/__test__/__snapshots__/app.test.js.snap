// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render login properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchLoggedUser={[Function]}
    loggedUser={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <AppBar
          iconClassNameRight="muidocs-icon-navigation-expand-more"
          onLeftIconButtonClick={[Function]}
          showMenuIconButton={false}
          title="Spotifood"
          zDepth={1}
/>,
        <Login />,
      ],
      "className": "filter-open",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "iconClassNameRight": "muidocs-icon-navigation-expand-more",
          "onLeftIconButtonClick": [Function],
          "showMenuIconButton": false,
          "title": "Spotifood",
          "zDepth": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <AppBar
            iconClassNameRight="muidocs-icon-navigation-expand-more"
            onLeftIconButtonClick={[Function]}
            showMenuIconButton={false}
            title="Spotifood"
            zDepth={1}
/>,
          <Login />,
        ],
        "className": "filter-open",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "iconClassNameRight": "muidocs-icon-navigation-expand-more",
            "onLeftIconButtonClick": [Function],
            "showMenuIconButton": false,
            "title": "Spotifood",
            "zDepth": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
